Build failure: immich

## Steps To Reproduce

Steps to reproduce the behavior:

1. build `immich`

```
nix build github:nixos/nixpkgs#immich
```

## Build log

<!-- insert build log in code block in collapsable section -->

<details>

<summary>Build Log</summary>

```
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/vj3r8lf1f3nfrb2bp08nn7yymzchxb5d-source/server
source root is server
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Executing npmConfigHook
Configuring npm
Validating consistency between /build/server/package-lock.json and /nix/store/wqrakixsc4axf3k56w0fnshf2nig38cr-immich-1.123.0-npm-deps/package-lock.json
Making cache writable
[1G[0K\[1G[0K[1G[0K\[1G[0K[1G[0KInstalling dependencies
[1mnpm[22m [33mwarn[39m [94mdeprecated[39m glob@8.1.0: Glob versions prior to v9 are no longer supported
[1mnpm[22m [33mwarn[39m [94mdeprecated[39m text-encoding@0.6.4: no longer maintained

added 1166 packages, and audited 1167 packages in 13s

181 packages are looking for funding
  run `npm fund` for details

found [32m[1m0[22m[39m vulnerabilities
patching script interpreter paths in node_modules
node_modules/geobuf/bin/shp2geobuf: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/geobuf/bin/json2geobuf: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/geobuf/bin/geobuf2json: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/typescript/bin/tsc: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/typescript/bin/tsserver: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/regexp-tree/bin/regexp-tree: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/handlebars/bin/handlebars: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/regjsparser/bin/parser: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/regjsparser/node_modules/jsesc/bin/jsesc: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/eslint/bin/eslint.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/webpack/bin/webpack.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/marked/bin/marked.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/nopt/bin/nopt.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/js-beautify/js/src/cli.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/js-beautify/js/lib/cli.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/js-beautify/js/bin/js-beautify.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/js-beautify/js/bin/html-beautify.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/js-beautify/js/bin/css-beautify.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/js-beautify/node_modules/nopt/bin/nopt.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/tailwindcss/lib/cli.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/uuid/dist/bin/uuid: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/tar/node_modules/mkdirp/bin/cmd.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/terser/bin/terser: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/terser/bin/uglifyjs: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/cpu-features/deps/cpu_features/scripts/run_integration.sh: interpreter directive changed from "#!/usr/bin/env bash" to "/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash"
node_modules/cpu-features/deps/cpu_features/scripts/test_integration.sh: interpreter directive changed from "#!/usr/bin/env bash" to "/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash"
node_modules/cpu-features/deps/cpu_features/scripts/make_release.sh: interpreter directive changed from "#!/usr/bin/env bash" to "/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash"
node_modules/cpu-features/deps/cpu_features/cmake/ci/doc/generate_image.sh: interpreter directive changed from "#!/usr/bin/env bash" to "/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash"
node_modules/acorn/bin/acorn: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/sucrase/bin/sucrase-node: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/sucrase/bin/sucrase: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/rimraf/dist/esm/bin.mjs: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/rimraf/node_modules/glob/dist/esm/bin.mjs: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/js-yaml/bin/js-yaml.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/esprima/bin/esvalidate.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/esprima/bin/esparse.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/tree-kill/cli.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/vite/bin/vite.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/he/bin/he: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/jsesc/bin/jsesc: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/ts-node/dist/bin-esm.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/ts-node/dist/bin-transpile.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/ts-node/dist/bin-script-deprecated.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/ts-node/dist/bin-cwd.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/ts-node/dist/bin.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/ts-node/dist/bin-script.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/react-email/dist/cli/index.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/react-email/node_modules/esbuild/bin/esbuild: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/react-email/node_modules/glob/dist/cjs/src/bin.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/which/bin/node-which: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/make-dir/node_modules/semver/bin/semver.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/cssesc/bin/cssesc: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/fluent-ffmpeg/node_modules/which/bin/which: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/msgpackr-extract/bin/download-prebuilds.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/nan/tools/1to2.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/import-in-the-middle/test/fixtures/executable: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/ua-parser-js/script/cli.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/node-gyp-build-optional-packages/bin.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/node-gyp-build-optional-packages/build-test.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/node-gyp-build-optional-packages/optional.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/typeorm/cli.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/typeorm/cli-ts-node-esm.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/typeorm/cli-ts-node-commonjs.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/typeorm/browser/cli-ts-node-esm.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/typeorm/browser/cli-ts-node-commonjs.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/typeorm/node_modules/mkdirp/dist/cjs/src/bin.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/prettier/bin/prettier.cjs: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/mime/cli.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/mime/src/build.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/@nuxtjs/opencollective/bin/opencollective.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/eslint-config-prettier/bin/cli.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/uglify-js/bin/uglifyjs: interpreter directive changed from "#! /usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/vitest/vitest.mjs: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/semver/bin/semver.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/browserslist/cli.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/rollup/dist/bin/rollup: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/bcrypt/test-docker.sh: interpreter directive changed from "#!/bin/sh" to "/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/sh"
node_modules/bcrypt/node_modules/node-addon-api/tools/conversion.js: interpreter directive changed from "#! /usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/pbf/bin/pbf: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/crc-32/bin/crc32.njs: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/@grpc/proto-loader/build/bin/proto-loader-gen-types.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/tsconfck/bin/tsconfck.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/update-browserslist-db/cli.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/sql-formatter/bin/sql-formatter-cli.cjs: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/properties-reader/node_modules/mkdirp/bin/cmd.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/next/dist/bin/next: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/systeminformation/lib/cli.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/cli-highlight/bin/highlight: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/@angular-devkit/schematics-cli/bin/schematics.d.ts: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/@angular-devkit/schematics-cli/bin/schematics.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/nearley/bin/nearleyc.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/nearley/bin/nearley-railroad.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/nearley/bin/nearley-test.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/nearley/bin/nearley-unparse.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/@nestjs/schedule/node_modules/uuid/dist/bin/uuid: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/@nestjs/typeorm/.husky/commit-msg: interpreter directive changed from "#!/bin/sh" to "/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/sh"
node_modules/@nestjs/typeorm/.husky/pre-commit: interpreter directive changed from "#!/bin/sh" to "/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/sh"
node_modules/@nestjs/cli/bin/nest.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/@nestjs/cli/node_modules/glob/dist/esm/bin.mjs: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/@nestjs/mapped-types/.husky/commit-msg: interpreter directive changed from "#!/bin/sh" to "/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/sh"
node_modules/@nestjs/mapped-types/.husky/pre-commit: interpreter directive changed from "#!/bin/sh" to "/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/sh"
node_modules/@babel/template/node_modules/@babel/parser/bin/babel-parser.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/@babel/generator/node_modules/jsesc/bin/jsesc: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/@babel/core/node_modules/semver/bin/semver.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/@babel/traverse/node_modules/@babel/parser/bin/babel-parser.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/@babel/helper-compilation-targets/node_modules/semver/bin/semver.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/@babel/parser/bin/babel-parser.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/color-support/bin.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/loose-envify/cli.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/json5/lib/cli.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/esbuild/bin/esbuild: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/utimes/node_modules/node-addon-api/tools/conversion.js: interpreter directive changed from "#! /usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/mkdirp/bin/cmd.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/shapefile/bin/shp2json: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/shapefile/bin/dbf2json: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/magicast/node_modules/@babel/parser/bin/babel-parser.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/jiti/bin/jiti.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/why-is-node-running/cli.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/resolve/bin/resolve: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/@mapbox/node-pre-gyp/bin/node-pre-gyp: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/@mapbox/node-pre-gyp/node_modules/rimraf/bin.js: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/normalize-package-data/node_modules/semver/bin/semver: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/glob/dist/esm/bin.mjs: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/nanoid/bin/nanoid.cjs: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/editorconfig/bin/editorconfig: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/vite-node/vite-node.mjs: interpreter directive changed from "#!/usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
node_modules/sha.js/bin.js: interpreter directive changed from "#! /usr/bin/env node" to "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node"
[1mnpm[22m [31merror[39m [94mcode[39m [33m1[39m
[1mnpm[22m [31merror[39m [94mpath[39m /build/server/node_modules/cpu-features
[1mnpm[22m [31merror[39m command failed
[1mnpm[22m [31merror[39m [94mcommand[39m sh -c node buildcheck.js > buildcheck.gypi && node-gyp rebuild
[1mnpm[22m [31merror[39m make: Entering directory '/build/server/node_modules/cpu-features/build'
[1mnpm[22m [31merror[39m   CC(target) Release/obj.target/cpu_features/deps/cpu_features/src/impl_aarch64_linux_or_android.o
[1mnpm[22m [31merror[39m   CC(target) Release/obj.target/cpu_features/deps/cpu_features/src/impl_aarch64_macos_or_iphone.o
[1mnpm[22m [31merror[39m   CC(target) Release/obj.target/cpu_features/deps/cpu_features/src/impl_arm_linux_or_android.o
[1mnpm[22m [31merror[39m   CC(target) Release/obj.target/cpu_features/deps/cpu_features/src/impl_mips_linux_or_android.o
[1mnpm[22m [31merror[39m   CC(target) Release/obj.target/cpu_features/deps/cpu_features/src/impl_ppc_linux.o
[1mnpm[22m [31merror[39m   CC(target) Release/obj.target/cpu_features/deps/cpu_features/src/impl_x86_freebsd.o
[1mnpm[22m [31merror[39m   CC(target) Release/obj.target/cpu_features/deps/cpu_features/src/impl_x86_linux_or_android.o
[1mnpm[22m [31merror[39m   CC(target) Release/obj.target/cpu_features/deps/cpu_features/src/impl_x86_macos.o
[1mnpm[22m [31merror[39m   CC(target) Release/obj.target/cpu_features/deps/cpu_features/src/impl_x86_windows.o
[1mnpm[22m [31merror[39m   CC(target) Release/obj.target/cpu_features/deps/cpu_features/src/filesystem.o
[1mnpm[22m [31merror[39m   CC(target) Release/obj.target/cpu_features/deps/cpu_features/src/stack_line_reader.o
[1mnpm[22m [31merror[39m   CC(target) Release/obj.target/cpu_features/deps/cpu_features/src/string_view.o
[1mnpm[22m [31merror[39m rm -f Release/obj.target/deps/cpu_features/cpu_features.a Release/obj.target/deps/cpu_features/cpu_features.a.ar-file-list; mkdir -p `dirname Release/obj.target/deps/cpu_features/cpu_features.a`
[1mnpm[22m [31merror[39m ar crs Release/obj.target/deps/cpu_features/cpu_features.a @Release/obj.target/deps/cpu_features/cpu_features.a.ar-file-list
[1mnpm[22m [31merror[39m   COPY Release/cpu_features.a
[1mnpm[22m [31merror[39m   CXX(target) Release/obj.target/cpufeatures/src/binding.o
[1mnpm[22m [31merror[39m make: Leaving directory '/build/server/node_modules/cpu-features/build'
[1mnpm[22m [31merror[39m gyp info it worked if it ends with ok
[1mnpm[22m [31merror[39m gyp info using node-gyp@10.2.0
[1mnpm[22m [31merror[39m gyp info using node@22.11.0 | linux | x64
[1mnpm[22m [31merror[39m gyp info find Python using Python version 3.12.8 found at "/nix/store/c9m6yd8fg1flz2j5r4bif1ib5j20a0cy-python3-3.12.8/bin/python3"
[1mnpm[22m [31merror[39m gyp WARN read config.gypi ENOENT: no such file or directory, open '/nix/store/qwrzxkqmsspprgq9s5j8bcr45lkign4k-nodejs-22.11.0-source/include/node/config.gypi'
[1mnpm[22m [31merror[39m gyp info spawn /nix/store/c9m6yd8fg1flz2j5r4bif1ib5j20a0cy-python3-3.12.8/bin/python3
[1mnpm[22m [31merror[39m gyp info spawn args [
[1mnpm[22m [31merror[39m gyp info spawn args '/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
[1mnpm[22m [31merror[39m gyp info spawn args 'binding.gyp',
[1mnpm[22m [31merror[39m gyp info spawn args '-f',
[1mnpm[22m [31merror[39m gyp info spawn args 'make',
[1mnpm[22m [31merror[39m gyp info spawn args '-I',
[1mnpm[22m [31merror[39m gyp info spawn args '/build/server/node_modules/cpu-features/build/config.gypi',
[1mnpm[22m [31merror[39m gyp info spawn args '-I',
[1mnpm[22m [31merror[39m gyp info spawn args '/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
[1mnpm[22m [31merror[39m gyp info spawn args '-I',
[1mnpm[22m [31merror[39m gyp info spawn args '/nix/store/qwrzxkqmsspprgq9s5j8bcr45lkign4k-nodejs-22.11.0-source/common.gypi',
[1mnpm[22m [31merror[39m gyp info spawn args '-Dlibrary=shared_library',
[1mnpm[22m [31merror[39m gyp info spawn args '-Dvisibility=default',
[1mnpm[22m [31merror[39m gyp info spawn args '-Dnode_root_dir=/nix/store/qwrzxkqmsspprgq9s5j8bcr45lkign4k-nodejs-22.11.0-source',
[1mnpm[22m [31merror[39m gyp info spawn args '-Dnode_gyp_dir=/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/lib/node_modules/npm/node_modules/node-gyp',
[1mnpm[22m [31merror[39m gyp info spawn args '-Dnode_lib_file=/nix/store/qwrzxkqmsspprgq9s5j8bcr45lkign4k-nodejs-22.11.0-source/$(Configuration)/node.lib',
[1mnpm[22m [31merror[39m gyp info spawn args '-Dmodule_root_dir=/build/server/node_modules/cpu-features',
[1mnpm[22m [31merror[39m gyp info spawn args '-Dnode_engine=v8',
[1mnpm[22m [31merror[39m gyp info spawn args '--depth=.',
[1mnpm[22m [31merror[39m gyp info spawn args '--no-parallel',
[1mnpm[22m [31merror[39m gyp info spawn args '--generator-output',
[1mnpm[22m [31merror[39m gyp info spawn args 'build',
[1mnpm[22m [31merror[39m gyp info spawn args '-Goutput_dir=.'
[1mnpm[22m [31merror[39m gyp info spawn args ]
[1mnpm[22m [31merror[39m gyp info spawn make
[1mnpm[22m [31merror[39m gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
[1mnpm[22m [31merror[39m In file included from ../src/binding.cc:3:
[1mnpm[22m [31merror[39m ../../nan/nan.h: In function 'void Nan::SetAccessor(v8::Local<v8::ObjectTemplate>, v8::Local<v8::String>, GetterCallback, SetterCallback, v8::Local<v8::Value>, v8::AccessControl, v8::PropertyAttribute, imp::Sig)':
[1mnpm[22m [31merror[39m ../../nan/nan.h:2548:19: error: no matching function for call to 'v8::ObjectTemplate::SetAccessor(v8::Local<v8::String>&, void (*&)(v8::Local<v8::Name>, const v8::PropertyCallbackInfo<v8::Value>&), void (*&)(v8::Local<v8::Name>, v8::Local<v8::Value>, const v8::PropertyCallbackInfo<void>&), v8::Local<v8::Object>&, v8::AccessControl&, v8::PropertyAttribute&)'
[1mnpm[22m [31merror[39m  2548 |   tpl->SetAccessor(
[1mnpm[22m [31merror[39m       |   ~~~~~~~~~~~~~~~~^
[1mnpm[22m [31merror[39m  2549 |       name
[1mnpm[22m [31merror[39m       |       ~~~~         
[1mnpm[22m [31merror[39m  2550 |     , getter_
[1mnpm[22m [31merror[39m       |     ~~~~~~~~~      
[1mnpm[22m [31merror[39m  2551 |     , setter_
[1mnpm[22m [31merror[39m       |     ~~~~~~~~~      
[1mnpm[22m [31merror[39m  2552 |     , obj
[1mnpm[22m [31merror[39m       |     ~~~~~          
[1mnpm[22m [31merror[39m  2553 |     , settings
[1mnpm[22m [31merror[39m       |     ~~~~~~~~~~     
[1mnpm[22m [31merror[39m  2554 |     , attribute
[1mnpm[22m [31merror[39m       |     ~~~~~~~~~~~    
[1mnpm[22m [31merror[39m  2555 | #if (NODE_MODULE_VERSION < NODE_16_0_MODULE_VERSION)
[1mnpm[22m [31merror[39m       | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[1mnpm[22m [31merror[39m  2556 |     , signature
[1mnpm[22m [31merror[39m       |     ~~~~~~~~~~~    
[1mnpm[22m [31merror[39m  2557 | #endif
[1mnpm[22m [31merror[39m       | ~~~~~~             
[1mnpm[22m [31merror[39m  2558 |   );
[1mnpm[22m [31merror[39m       |   ~                
[1mnpm[22m [31merror[39m In file included from /nix/store/qwrzxkqmsspprgq9s5j8bcr45lkign4k-nodejs-22.11.0-source/deps/v8/include/v8-function.h:15,
[1mnpm[22m [31merror[39m                  from /nix/store/qwrzxkqmsspprgq9s5j8bcr45lkign4k-nodejs-22.11.0-source/deps/v8/include/v8.h:33,
[1mnpm[22m [31merror[39m                  from /nix/store/qwrzxkqmsspprgq9s5j8bcr45lkign4k-nodejs-22.11.0-source/src/node.h:73,
[1mnpm[22m [31merror[39m                  from ../src/binding.cc:1:
[1mnpm[22m [31merror[39m /nix/store/qwrzxkqmsspprgq9s5j8bcr45lkign4k-nodejs-22.11.0-source/deps/v8/include/v8-template.h:1049:8: note: candidate: 'void v8::ObjectTemplate::SetAccessor(v8::Local<v8::String>, v8::AccessorGetterCallback, v8::AccessorSetterCallback, v8::Local<v8::Value>, v8::PropertyAttribute, v8::SideEffectType, v8::SideEffectType)'
[1mnpm[22m [31merror[39m  1049 |   void SetAccessor(
[1mnpm[22m [31merror[39m       |        ^~~~~~~~~~~
[1mnpm[22m [31merror[39m /nix/store/qwrzxkqmsspprgq9s5j8bcr45lkign4k-nodejs-22.11.0-source/deps/v8/include/v8-template.h:1052:61: note:   no known conversion for argument 5 from 'v8::AccessControl' to 'v8::PropertyAttribute'
[1mnpm[22m [31merror[39m  1052 |       Local<Value> data = Local<Value>(), PropertyAttribute attribute = None,
[1mnpm[22m [31merror[39m       |                                           ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~
[1mnpm[22m [31merror[39m /nix/store/qwrzxkqmsspprgq9s5j8bcr45lkign4k-nodejs-22.11.0-source/deps/v8/include/v8-template.h:1055:8: note: candidate: 'void v8::ObjectTemplate::SetAccessor(v8::Local<v8::Name>, v8::AccessorNameGetterCallback, v8::AccessorNameSetterCallback, v8::Local<v8::Value>, v8::PropertyAttribute, v8::SideEffectType, v8::SideEffectType)'
[1mnpm[22m [31merror[39m  1055 |   void SetAccessor(
[1mnpm[22m [31merror[39m       |        ^~~~~~~~~~~
[1mnpm[22m [31merror[39m /nix/store/qwrzxkqmsspprgq9s5j8bcr45lkign4k-nodejs-22.11.0-source/deps/v8/include/v8-template.h:1058:61: note:   no known conversion for argument 5 from 'v8::AccessControl' to 'v8::PropertyAttribute'
[1mnpm[22m [31merror[39m  1058 |       Local<Value> data = Local<Value>(), PropertyAttribute attribute = None,
[1mnpm[22m [31merror[39m       |                                           ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~
[1mnpm[22m [31merror[39m ../../nan/nan.h: In function 'void Nan::SetAccessor(v8::Local<v8::ObjectTemplate>, v8::Local<v8::String>, GetterCallback, SetterCallback, v8::Local<v8::Value>, v8::AccessControl, v8::PropertyAttribute)':
[1mnpm[22m [31merror[39m ../../nan/nan.h:2594:19: error: no matching function for call to 'v8::ObjectTemplate::SetAccessor(v8::Local<v8::String>&, void (*&)(v8::Local<v8::Name>, const v8::PropertyCallbackInfo<v8::Value>&), void (*&)(v8::Local<v8::Name>, v8::Local<v8::Value>, const v8::PropertyCallbackInfo<void>&), v8::Local<v8::Object>&, v8::AccessControl&, v8::PropertyAttribute&)'
[1mnpm[22m [31merror[39m  2594 |   tpl->SetAccessor(
[1mnpm[22m [31merror[39m       |   ~~~~~~~~~~~~~~~~^
[1mnpm[22m [31merror[39m  2595 |       name
[1mnpm[22m [31merror[39m       |       ~~~~         
[1mnpm[22m [31merror[39m  2596 |     , getter_
[1mnpm[22m [31merror[39m       |     ~~~~~~~~~      
[1mnpm[22m [31merror[39m  2597 |     , setter_
[1mnpm[22m [31merror[39m       |     ~~~~~~~~~      
[1mnpm[22m [31merror[39m  2598 |     , obj
[1mnpm[22m [31merror[39m       |     ~~~~~          
[1mnpm[22m [31merror[39m  2599 |     , settings
[1mnpm[22m [31merror[39m       |     ~~~~~~~~~~     
[1mnpm[22m [31merror[39m  2600 |     , attribute
[1mnpm[22m [31merror[39m       |     ~~~~~~~~~~~    
[1mnpm[22m [31merror[39m  2601 |   );
[1mnpm[22m [31merror[39m       |   ~                
[1mnpm[22m [31merror[39m /nix/store/qwrzxkqmsspprgq9s5j8bcr45lkign4k-nodejs-22.11.0-source/deps/v8/include/v8-template.h:1049:8: note: candidate: 'void v8::ObjectTemplate::SetAccessor(v8::Local<v8::String>, v8::AccessorGetterCallback, v8::AccessorSetterCallback, v8::Local<v8::Value>, v8::PropertyAttribute, v8::SideEffectType, v8::SideEffectType)'
[1mnpm[22m [31merror[39m  1049 |   void SetAccessor(
[1mnpm[22m [31merror[39m       |        ^~~~~~~~~~~
[1mnpm[22m [31merror[39m /nix/store/qwrzxkqmsspprgq9s5j8bcr45lkign4k-nodejs-22.11.0-source/deps/v8/include/v8-template.h:1052:61: note:   no known conversion for argument 5 from 'v8::AccessControl' to 'v8::PropertyAttribute'
[1mnpm[22m [31merror[39m  1052 |       Local<Value> data = Local<Value>(), PropertyAttribute attribute = None,
[1mnpm[22m [31merror[39m       |                                           ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~
[1mnpm[22m [31merror[39m /nix/store/qwrzxkqmsspprgq9s5j8bcr45lkign4k-nodejs-22.11.0-source/deps/v8/include/v8-template.h:1055:8: note: candidate: 'void v8::ObjectTemplate::SetAccessor(v8::Local<v8::Name>, v8::AccessorNameGetterCallback, v8::AccessorNameSetterCallback, v8::Local<v8::Value>, v8::PropertyAttribute, v8::SideEffectType, v8::SideEffectType)'
[1mnpm[22m [31merror[39m  1055 |   void SetAccessor(
[1mnpm[22m [31merror[39m       |        ^~~~~~~~~~~
[1mnpm[22m [31merror[39m /nix/store/qwrzxkqmsspprgq9s5j8bcr45lkign4k-nodejs-22.11.0-source/deps/v8/include/v8-template.h:1058:61: note:   no known conversion for argument 5 from 'v8::AccessControl' to 'v8::PropertyAttribute'
[1mnpm[22m [31merror[39m  1058 |       Local<Value> data = Local<Value>(), PropertyAttribute attribute = None,
[1mnpm[22m [31merror[39m       |                                           ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~
[1mnpm[22m [31merror[39m ../src/binding.cc: At global scope:
[1mnpm[22m [31merror[39m /nix/store/qwrzxkqmsspprgq9s5j8bcr45lkign4k-nodejs-22.11.0-source/src/node.h:1232:7: warning: cast between incompatible function types from 'void (*)(Nan::ADDON_REGISTER_FUNCTION_ARGS_TYPE)' {aka 'void (*)(v8::Local<v8::Object>)'} to 'node::addon_register_func' {aka 'void (*)(v8::Local<v8::Object>, v8::Local<v8::Value>, void*)'} [-Wcast-function-type]
[1mnpm[22m [31merror[39m  1232 |       (node::addon_register_func) (regfunc),                          \
[1mnpm[22m [31merror[39m       |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[1mnpm[22m [31merror[39m /nix/store/qwrzxkqmsspprgq9s5j8bcr45lkign4k-nodejs-22.11.0-source/src/node.h:1266:3: note: in expansion of macro 'NODE_MODULE_X'
[1mnpm[22m [31merror[39m  1266 |   NODE_MODULE_X(modname, regfunc, NULL, 0)  // NOLINT (readability/null_usage)
[1mnpm[22m [31merror[39m       |   ^~~~~~~~~~~~~
[1mnpm[22m [31merror[39m ../src/binding.cc:151:1: note: in expansion of macro 'NODE_MODULE'
[1mnpm[22m [31merror[39m   151 | NODE_MODULE(cpufeatures, init)
[1mnpm[22m [31merror[39m       | ^~~~~~~~~~~
[1mnpm[22m [31merror[39m make: *** [cpufeatures.target.mk:117: Release/obj.target/cpufeatures/src/binding.o] Error 1
[1mnpm[22m [31merror[39m gyp ERR! build error 
[1mnpm[22m [31merror[39m gyp ERR! stack Error: `make` failed with exit code: 2
[1mnpm[22m [31merror[39m gyp ERR! stack at ChildProcess.<anonymous> (/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:216:23)
[1mnpm[22m [31merror[39m gyp ERR! System Linux 6.6.66
[1mnpm[22m [31merror[39m gyp ERR! command "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/bin/node" "/nix/store/fkyp1bm5gll9adnfcj92snyym524mdrj-nodejs-22.11.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
[1mnpm[22m [31merror[39m gyp ERR! cwd /build/server/node_modules/cpu-features
[1mnpm[22m [31merror[39m gyp ERR! node -v v22.11.0
[1mnpm[22m [31merror[39m gyp ERR! node-gyp -v v10.2.0
[1mnpm[22m [31merror[39m gyp ERR! not ok
[1mnpm[22m [31merror[39m A complete log of this run can be found in: /build/cache/_logs/2024-12-28T23_35_01_744Z-debug-0.log

```

</details>

## Additional context

<!-- Add any other context about the problem here. -->

The break was introduced to `master` from `staging-next` in #361878 (fc9c33366b98), which bumped the default `nodejs` version from 20 to 22. Diffing `nix derivation show` before (585f76290ed6) and after that merge shows no major differences aside from that change, and overriding the `nodejs` version fixes the build. The following fails:

```
nix-repl> :b let nixpkgs = builtins.getFlake "github:nixos/nixpkgs"; in with nixpkgs.legacyPackages.x86_64-linux; immich
```

but overriding to use nodejs_20 succeeds:

```
nix-repl> :b let nixpkgs = builtins.getFlake "github:nixos/nixpkgs"; in with nixpkgs.legacyPackages.x86_64-linux; immich.override { nodejs = nodejs_20; }
```

Note that upstream Immich updated from 20 to 22 https://github.com/immich-app/immich/commit/a28b92b3cc439a0de47bff8062126f254aa4ee26 in `v1.120.0`, so while we _could_ fix by pinning 20, we should probably figure out what will make 22 work.

## Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->

 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.66, NixOS, 24.11 (Vicuna), 24.11.20241221.dirty`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.10`
 - nixpkgs: `/nix/store/7lz01w5j2mw1nnshqmb6p47s38sia9pf-source`

## Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

@dotlambda
@jvanbruegge
@Scrumplex
@titaniumtown

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/im/immich/package.nix
+++ b/pkgs/by-name/im/immich/package.nix
@@ -187,7 +187,7 @@ buildNpmPackage' {
     mkdir node_modules
     ln -s ${node-addon-api} node_modules/node-addon-api
 
-    ${lib.getExe nodejs} install/check
+    node install/check
 
     rm -r node_modules
 

--- a/pkgs/by-name/im/immich/package.nix
+++ b/pkgs/by-name/im/immich/package.nix
@@ -5,7 +5,7 @@
   fetchFromGitHub,
   fetchpatch2,
   python3,
-  nodejs,
+  nodejs_20,
   node-gyp,
   runCommand,
   nixosTests,
@@ -27,6 +27,7 @@
   vips,
 }:
 let
+  nodejs = nodejs_20;
   buildNpmPackage' = buildNpmPackage.override { inherit nodejs; };
   sources = lib.importJSON ./sources.json;
   inherit (sources) version;

